third_party/txt/src/txt/fuchsia_font_manager.cc

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "txt/fuchsia_font_manager.h"
+
+#include <zx/vmar.h>
+
+#include "lib/fsl/vmo/sized_vmo.h"
+#include "lib/fxl/logging.h"
+#include "txt/asset_font_style_set.h"
+
+namespace txt {
+
+namespace {
+
+fonts::FontProviderPtr* g_font_provider = nullptr;
+
+fonts::FontProviderPtr& GetFontProvider() {
```

Comment:
Instead of making this a singleton here, lets move this into the application controller in the content handler.